tests/manage/z_cluster/test_multiple_mon_pods_stays_on_same_node.py

```diff
@@ -0,0 +1,215 @@
+import json
+import logging
+import pytest
+import time
+
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier4b,
+    skipif_ocs_version,
+    ignore_leftovers,
+)
+from ocs_ci.ocs.constants import (
+    ROOK_CEPH_MON_ENDPOINTS,
+    CONFIGMAP,
+    OPENSHIFT_STORAGE_NAMESPACE,
+    STATUS_RUNNING,
+    STATUS_PENDING,
+    MON_APP_LABEL,
+    DEPLOYMENT,
+    POD,
+    OPERATOR_LABEL,
+)
+from ocs_ci.ocs.exceptions import TimeoutExpiredError, ResourceWrongStatusException
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.node import get_mon_running_nodes
+from ocs_ci.ocs.resources.pod import (
+    get_mon_pods,
+    get_operator_pods,
+    get_pod_node,
+    delete_pods,
+)
+from ocs_ci.utility.utils import ceph_health_check
+
+log = logging.getLogger(__name__)
+POD_OBJ = OCP(kind=POD, namespace=OPENSHIFT_STORAGE_NAMESPACE)
+
+
+@tier4b
+@ignore_leftovers
+@skipif_ocs_version("<4.8")
+@pytest.mark.polarion_id("OCS-2593")
+@pytest.mark.bugzilla("1974204")
+class TestMultipleMonPodsStaysOnSameNode(ManageTest):
+    """
+    Verify multiple mon pods stays on same node
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Verifies cluster is healthy
+        """
+        mon_pod = get_mon_pods()
+
+        def finalizer():
+
+            try:
+
+                # Validate all mon pods are running
+                log.info("Validate all mons are up and running")
+                POD_OBJ.wait_for_resource(
+                    condition=STATUS_RUNNING,
+                    selector=MON_APP_LABEL,
+                    resource_count=len(mon_pod),
+                )
+                log.info("All mons are up and running")
+
+            except (TimeoutExpiredError, ResourceWrongStatusException) as ex:
+                log.error(f"{ex}")
+                # Restart operator
+                operator_pod_obj = get_operator_pods()
+                delete_pods(pod_objs=operator_pod_obj)
+
+                # Wait untill mon pod recovery
+                POD_OBJ.wait_for_resource(
+                    condition=STATUS_RUNNING,
+                    selector=MON_APP_LABEL,
+                    resource_count=len(mon_pod),
+                    timeout=3600,
+                    sleep=5,
+                )
+                log.info("All mons are up and running")
+
+                # Check the ceph health OK
+                ceph_health_check(tries=90, delay=15)
+
+        request.addfinalizer(finalizer)
+
+    def mon_pods_running_on_same_node(self):
```

Comment:
Can you please move this function from the test to some common place.

Reply:
moved to helpers.py